c7n/filters/core.py

```diff
@@ -406,10 +406,17 @@ def get_resource_value(self, k, i):
         if k.startswith('tag:'):
             tk = k.split(':', 1)[1]
             r = None
-            for t in i.get("Tags", []):
-                if t.get('Key') == tk:
-                    r = t.get('Value')
-                    break
+            if 'Tags' in i:
+                for t in i.get("Tags", []):
+                    if t.get('Key') == tk:
+                        r = t.get('Value')
+                        break
+            elif 'tags' in i:
+                tags = i.get("tags", [])
+                for key in tags:
+                    if key == tk:
+                        r = tags[key]
+                        break
```

Comment:
I think you can do this more concise, perhaps
```
            elif 'tags' in i:
                r = i.get("tags", {}).get(tk, None)
```

also we probably need @kapilt to chime in with his opinion on doing this at all.  I think it is a nice quick solution to Azure parity for a useful feature but it is also a hack on a hack and unsure what GCP looks like here, so his call.

Reply:
Just to provide more detail on @stefangordon's last comment, in the scenario where the policy is something like
```
filters:
  - "tag:CreatedBy" : present
```
I created an `AzureValueFilter(ValueFilter)` with an overridden `get_resource_value` method, but I wasn't able to get to the overridden `get_resource_value` without the `type` specified. Went straight to the original `ValueFilter`.

Reply:
Yeah would be trickier to change the default filter out I think.

Reply:
Fwiw, the value filter is ripe for some refactoring, i was checking on out cyclomatic complexity scores on the code base, and this particular class is lights up as overly high.  I'll have a deeper look at this pr/use case later today.

Reply:
Ok that makes sense - @tbarlow12 let's try to optimize that implementation per my first comment before we commit though, we don't want to implement our own `.get` as I'm sure it is slower than the real thing.